src/ewig/buffer.cpp

```diff
@@ -480,10 +480,10 @@ text selected_text(buffer buf)
                 : buf.content)
             .take(ends.row + 1)
             .drop(starts.row)
-            .update(ends.row-starts.row, [&] (auto l) {
+            .update(ends.row-starts.row, [&, ends = ends] (auto l) {
                 return l.take(line_char(l, ends.col));
             })
-            .update(0, [&] (auto l) {
+            .update(0, [&, starts = starts] (auto l) {
```

Comment:
I don''t understand why such capture is needed. It should work with the `&`. These lambdas do not exit the scope of the functions they are passed to, so the `[&]` is totally safe here.

Reply:
The error I got for all of these was:

`Building CXX object CMakeFiles/ewig.dir/src/ewig/buffer.cpp.o`
`/Users/..../projects/arximboldi/ewig/src/ewig/buffer.cpp:487:44: error: reference to local binding` `'starts' declared in enclosing function 'ewig::selected_text'`
`                return l.drop(line_char(l, starts.col));`
 `                                          ^`
`/Users/..../projects/arximboldi/ewig/src/ewig/buffer.cpp:472:11: note: 'starts' declared here`
`    auto [starts, ends] = selected_region(buf);`
`          ^`
( see also: https://stackoverflow.com/questions/46114214/lambda-implicit-capture-fails-with-variable-declared-from-structured-binding )

Reply:
I see, so it seems like an AppleClang "bug" (as the question notes, there is some ambiguity in the standard, but everybody agrees it is "meant" to work). It seems to work with LLVM Clang and Gcc, so I would ideally not merge this part of the change (the purpose of this code-base is educational.)

Reply:
I've tried a simple example in godbolt.org, its an issue of clang vs. gcc ( nothing apple specific ). clang, in none of its versions, allows automatic capturing of the structured bindings. gcc does. So I would argue specifically for education that it should be merged.

Reply:
Ok, I'm convinced :smile: